core/src/Revolution/modPhpThumb.php

```diff
@@ -302,10 +302,11 @@ public function RedirectToCachedFile()
             $this->SendSaveAsFileHeaderIfNeeded();
 
             header('Last-Modified: ' . gmdate('D, d M Y H:i:s', $nModified) . ' GMT');
-            if (@$_SERVER['HTTP_IF_MODIFIED_SINCE'] && ($nModified == strtotime($_SERVER['HTTP_IF_MODIFIED_SINCE'])) && @$_SERVER['SERVER_PROTOCOL']) {
-                header($_SERVER['SERVER_PROTOCOL'] . ' 304 Not Modified');
-                exit;
-            }
+            # Fix PHP warning: Undefined array key "HTTP_IF_MODIFIED_SINCE"
+               if (isset($_SERVER['HTTP_IF_MODIFIED_SINCE']) && ($nModified == strtotime($_SERVER['HTTP_IF_MODIFIED_SINCE'])) && isset($_SERVER['SERVER_PROTOCOL'])) {
```

Comment:
You can simplify this just a bit, consider:

```suggestion

            if (isset($_SERVER['HTTP_IF_MODIFIED_SINCE'], $_SERVER['SERVER_PROTOCOL']) && $nModified == strtotime($_SERVER['HTTP_IF_MODIFIED_SINCE'])) {
```